pkg/controller/transaction/controller.go

```diff
@@ -86,30 +86,38 @@ func (r *Reconciler) Reconcile(id controller.ID) (controller.Result, error) {
 	}
 
 	log.Infof("Reconciling transaction %v", transaction)
-	switch t := transaction.Transaction.(type) {
-	case *configapi.Transaction_Change:
-		if ok, err := r.reconcileTransactionChange(ctx, transaction, t.Change); err != nil {
-			log.Warnf("Failed to reconcile transaction: %s, %s", transactionID, err)
-			return controller.Result{}, err
-		} else if ok {
-			return controller.Result{}, nil
-		}
-	case *configapi.Transaction_Rollback:
-		if ok, err := r.reconcileTransactionRollback(ctx, transaction, t.Rollback); err != nil {
-			log.Warnf("Failed to reconcile transaction: %s, %s", transactionID, err)
-			return controller.Result{}, err
-		} else if ok {
+
+	// If the transaction is in a completed state, queue the next transaction to be reconciled
+	if transaction.Status.State == configapi.TransactionState_TRANSACTION_COMPLETE ||
+		transaction.Status.State == configapi.TransactionState_TRANSACTION_FAILED {
+		nextTransaction, err := r.transactions.GetByIndex(ctx, transaction.Index+1)
+		if err != nil {
+			if !errors.IsNotFound(err) {
+				return controller.Result{}, err
+			}
 			return controller.Result{}, nil
 		}
+		log.Debugf("Queueing next transaction %d", nextTransaction.Index)
+		return controller.Result{Requeue: controller.ID{Value: nextTransaction.ID}}, nil
```

Comment:
If one transaction is complete, we should always reconcile the next transaction in case completing this transaction unblocked the next one.